MSBUILD Server dogfood

SimaTian opened the pull request:
### MSBUILD Server dogfood
Part of [#9379](https://github.com/dotnet/msbuild/issues/9379)

Please help us with our dofgooding effort for the MSBuild server feature.
For now we've dogfooded internally for our pipelines, we're running without issues within Aspnetcore and we would like to continue the rollout before we're confident enough to make Server build the new default.

SimaTian commented:
Note: I'm not familiar with your pipelines. Turning on MSBUILD server is done via setting the environment variable MSBUILSUSESERVER to 1 so that is what I was trying to achieve.
If there is a better place to do it, please let me know.

ViktorHofer commented:
@jkoritzinsky where's the best place to set an env var in runtime's YML that should impact all msbuild invocations?

jkoritzinsky commented:
I'd put it in global-build-step.yml.

That will catch almost all product builds (not runtime tests).

SimaTian committed:
updating based on review with Jemery Koritzinsky

janvorli commented on eng/build.ps1:
```diff
@@ -262,6 +262,10 @@ if ($vs) {
   $env:VSDebugger_ValidateDotnetDebugLibSignatures=0;
 
+  # MSBuild Server dogfood.
+  $env:DOTNET_CLI_USE_MSBUILD_SERVER=1;
```
Do we really want to set it without a way to opt out? What if we hit a problem with the server that would prevent local builds from succeeding and all developers would get affected?

janvorli replied:
That is not as simple when you often build different branches, which I do during development. Editing a file for each branch, paying attention to not to accidentally add the change to your change list etc. sounds like a pain. 
The way to opt out can be as simple as not setting the DOTNET_CLI_USE_MSBUILD_SERVER if it was already set. Then in case there are some issues, I can set that variable to 0 until the issue is resolved. I can have that set globally so all shells I run would pick it.

SimaTian committed:
Clean up environment variable definitions

Removed redundant environment variable definitions for MSBUILDUSESERVER and DOTNET_CLI_USE_MSBUILD_SERVER.

SimaTian committed:
adding an escape hatch for the dogfood

SimaTian committed:
Merge branch 'MSBUILD-dogfood' of https://github.com/dotnet/runtime into MSBUILD-dogfood

SimaTian committed:
minor cleanup

SimaTian committed:
Fix environment variable name for MSBuild server

SimaTian committed:
affordance for old powershell

SimaTian committed:
Merge branch 'MSBUILD-dogfood' of https://github.com/dotnet/runtime into MSBUILD-dogfood
